File: problem-solving/baekjoon/10845_큐.go
// https://www.acmicpc.net/problem/10845
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscanln(reader, &n)

	queue := []int{}
	for i := 0; i < n; i++ {
		var command string
		var number int
		fmt.Fscanln(reader, &command, &number)

		switch command {
		case "push":
			queue = append(queue, number)
		case "pop":
			output := -1
			if len(queue) > 0 {
				output = queue[0]
				queue = queue[1:]
			}
			fmt.Fprintln(writer, output)
		case "size":
			fmt.Fprintln(writer, len(queue))
		case "empty":
			if len(queue) == 0 {
				fmt.Fprintln(writer, 1)
			} else {
				fmt.Fprintln(writer, 0)
			}
		case "front":
			output := -1
			if len(queue) > 0 {
				output = queue[0]
			}
			fmt.Fprintln(writer, output)
		case "back":
			output := -1
			if len(queue) > 0 {
				output = queue[len(queue)-1]
			}
			fmt.Fprintln(writer, output)
		}
	}
}

File: problem-solving/baekjoon/10845_큐.py
# https://www.acmicpc.net/problem/10845
import sys

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    queue = []
    for i in range(n):
        inputs = sys.stdin.readline().split()
        command = inputs[0]
        number = 0
        if len(inputs) == 2:
            number = inputs[1]
        
        if command == "push":
            queue.append(number)
        elif command == "pop":
            output = -1
            if len(queue) > 0:
                output = queue[0]
                queue.pop(0)
            print(output)
        elif command == "size":
            print(len(queue))
        elif command == "empty":
            if len(queue) == 0:
                print(1)
            else:
                print(0)
        elif command == "front":
            output = -1
            if len(queue) > 0:
                output = queue[0]
            print(output)
        elif command == "back":
            output = -1
            if len(queue) > 0:
                output = queue[len(queue)-1]
            print(output)